common/app/views/fragments/contentImage.scala.html

```diff
@@ -13,9 +13,10 @@
          class="gu-image"
          itemprop="contentUrl">
 
-    @image.caption.map { caption =>
-        <figcaption itemprop="description" class="caption caption--img">
-            @Html(caption) @if(image.displayCredit && !image.credit.exists(caption.endsWith)) { @image.credit.map(Html(_)) }
+    @if(image.caption.nonEmpty || (image.displayCredit && image.credit.nonEmpty)) {
+        <figcaption class="caption caption--img" itemprop="description">
+            @image.caption.map(Html(_))
+            @if(image.displayCredit && !image.credit.exists(image.caption.getOrElse("").endsWith)) { @image.credit.map(Html(_)) }
```

Comment:
This is a bit hard to read. You could move it into a helper function in a pure Scala file (see stuff in `views.support` for example) and then import that here:

```scala
def creditEndsWithCaption(image: Image) = for {
    credit <- image.credit
    caption <- image.caption
} yield caption.endsWith(credit)
```